advanced/wallets/react-wallet-v2/src/utils/SuiRequestHandlerUtil.ts

```diff
@@ -0,0 +1,63 @@
+import { getWallet } from '@/utils/SuiWalletUtil'
+import { getSignParamsMessage } from '@/utils/HelperUtil'
+import { formatJsonRpcError, formatJsonRpcResult } from '@json-rpc-tools/utils'
+import { SignClientTypes } from '@walletconnect/types'
+import { getSdkError } from '@walletconnect/utils'
+import SettingsStore from '@/store/SettingsStore'
+import { SUI_SIGNING_METHODS } from '@/data/SuiData'
+
+type RequestEventArgs = Omit<SignClientTypes.EventArguments['session_request'], 'verifyContext'>
+
+export async function approveSuiRequest(requestEvent: RequestEventArgs) {
+  const { params, id } = requestEvent
+  const { chainId, request } = params
+
+  SettingsStore.setActiveChainId(chainId)
+
+  const wallet = await getWallet()
+
+  switch (request.method) {
+    case SUI_SIGNING_METHODS.SUI_SIGN_PERSONAL_MESSAGE:
+      try {
+        const message = request.params.message
+        const signedMessage = await wallet.signMessage({ message })
+        return formatJsonRpcResult(id, signedMessage)
+      } catch (error: any) {
+        console.error(error)
+        alert(error.message)
+        return formatJsonRpcError(id, error.message)
+      }
+    case SUI_SIGNING_METHODS.SUI_SIGN_TRANSACTION:
+      try {
+        const result = await wallet.signTransaction({
```

Comment:
For signTransaction we need transaction bytes in the response to calculate digest in the SDK

Reply:
Bumping @ganchoradkov